Automattic/wp-calypso
# Comments: Split the Tree and List components

Depends on #21112, #21115, #21121
Replaces #21077

#21004 and #21035 introduced a new `ui.comments.queries` state that helps handling the Comments Management pagination.

To improve this section performance, we decided to use a combination of the comment list and the new comment count endpoints, instead of the comment tree one.
To keep supporting all our cases (e.g. we have a threaded view behind feature flag that requires the tree approach), I've moved the tree approach into a new `CommentTree` component, and modified the old `CommentList` to only use the list+count approach.

The `CommentsManagement` Main component will decide which one to render, or *not to* if Jetpack is not up to date or the user has not enough permissions.

The `CommentTree` component is in fact a carbon copy of the old `CommentList`, with just the `sortOrder` and `setSortOrder` props renamed as `order` and `sortOrder`.
There should be no need to review `my-sites/comments/comment-tree/index.jsx`.

## Testing instructions

- Does it use `CommentList`?
  - Open `/comments` on this branch and navigate around.
  - Make sure the component used to render the list is `CommentList` (visually: it loads all the comments at once, instead of trickling in one by one).
- Does it ask to update Jetpack if the site uses a version < 5.6?
  - Open `/comments` on a totally outdated Jetpack site.
  - You should see an illustration prompting to update Jetpack, and of course no comments should be displayed.
- On `CommentList`, make sure all these actions work as intended:
  - Change tab
  - Change sort order
  - Change page
  - Bulk change status and permanent delete
  - (Single) change status and permanent delete
  - Approve and reply

The following test case is only to check that things still work with the old comments tree approach, in the one single case we still need to use it (and which is not enabled in any environment)
We know it's not good performance-wise, hence the whole rework.
Also, its design is still WIP.

- Does it use `CommentTree` when in threaded view?
  - Load Calypso with the `?flags=comments/management/threaded-view` query parameter to enable the threaded view.
  - Open `/comments` and navigate to a post view.
  - Make sure the component used to render the list is `CommentTree` (visually: it trickles in the comments one by one, instead of loading all of them at once).
  - Make sure all these actions still work as intended:
    - Change tab
    - Change sort order
    - Change page
    - Bulk change status and permanent delete
    - (Single) change status and permanent delete
    - Approve and reply
